contracts/managers/context/ContextManager.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.26;
+
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {AccessManagedUpgradeable} from "../access/AccessManagedUpgradeable.sol";
+import {ContextManagerStorageLib} from "./ContextManagerStorageLib.sol";
+import {IContextClient} from "./IContextClient.sol";
+
+struct ExecuteData {
+    address[] addrs;
+    bytes[] data;
+}
+
+struct ContextDataWithSender {
+    address sender;
+    uint256 expirationTime;
```

Comment:
maybe chainId also?

Reply:
The chainId is embedded in the signed message and in the contract, so there is no need for an additional variable as it would be unused during verification.